lib/Onyx.js

```diff
@@ -1248,19 +1259,34 @@ function mergeCollection(collectionKey, collection) {
 
     return getAllKeys()
         .then((persistedKeys) => {
-            // Split to keys that exist in storage and keys that don't
-            const [existingKeys, newKeys] = _.chain(collection)
-                .keys()
-                .partition(key => persistedKeys.includes(key))
-                .value();
+            const existingKeysToKeep = [];
+            const keysToRemove = [];
+            const collectionWithoutKeysToRemove = collection;
+            const keyValuePairsForExistingCollection = [];
+            const keyValuePairsForNewCollection = [];
+            _.each(collection, (value, key) => {
+                if (value === null) {
+                    keysToRemove.push(key);
+                    delete collectionWithoutKeysToRemove[key];
```

Comment:
Rather than delete the keys it might be safer to add the ones we are keeping to a new object and avoid mutating the object passed to `mergeCollection()` (I think maybe we are mutating it).

Reply:
This makes sense to me